types/email-templates/index.d.ts

```diff
@@ -103,11 +112,26 @@ declare namespace EmailTemplate {
          * @param view The Html pug to render
          * @param locals The template Variables
          */
-        function render(view: string, locals: any): Promise<string>;
+        function render(view: string, locals?: any): Promise<string>;
+
+        /**
+         * Render all available template files for a given email
+         * template (e.g. `html.pug`, `text.pug`, and `subject.pug`)
+         *
+         * @param view Name of the template
+         * @param locals The template variables
+         */
+        function renderAll(view: string, locals?: any): Promise<EmailMessage>;
 
         /**
          * Send the Email
          */
         function send(options: EmailOptions): any;
```

Comment:
This should return the EmailMessage as well

Reply:
Looking at this part, it's rather a `Promise` of `SentMessageInfo` from nodemailer (which is currently just defined as `any` in the nodemailer typings) and **not** the newly defined interface `EmailMessage`.

See:  https://github.com/niftylettuce/email-templates/blob/af01d14fef9df70370f82761d7116ccb979edc2c/src/index.js#L292

I'll thus change it to return `Promise<any>` instead.